### PR TITLE
Generic/LowercasedFilename: improve code coverage

### PR DESCRIPTION
**Pull request by rodrigoprimo**
# Description

This PR improves code coverage for the `Generic.Files.LowercasedFilename` sniff. I had to create a test file that does not follow the naming convention used by PHPCS to test how the sniff behaves when handling a file that only contains lowercase letters. To improve the readability of this file, I also used underscores.

## Related issues/external references

Part of https://github.com/PHPCSStandards/PHP_CodeSniffer/issues/146
Issue found while working on this PR: https://github.com/PHPCSStandards/PHP_CodeSniffer/issues/682

## Types of changes

- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_
    - [ ] This change is only breaking for integrators, not for external standards or end-users.
- [ ] Documentation improvement

## PR checklist

- [x] I have checked there is no other PR open for the same change.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md).
- [x] I grant the project the right to include and distribute the code under the BSD-3-Clause license (and I have the right to grant these rights).
- [ ] I have added tests to cover my changes.
- [x] I have verified that the code complies with the projects coding standards.
- [ ] \[Required for new sniffs\] I have added XML documentation for the sniff.

### CODE REVIEW

File: src/Standards/Generic/Sniffs/Files/LowercasedFilenameSniff.php [hunk 1]
```diff
@@ -44,7 +44,7 @@ public function process(File $phpcsFile, $stackPtr)
     {
         $filename = $phpcsFile->getFilename();
         if ($filename === 'STDIN') {
-            return $phpcsFile->numTokens;
+            return $phpcsFile->numTokens; // @codeCoverageIgnore
```
Comment (jrfnl) [inline src/Standards/Generic/Sniffs/Files/LowercasedFilenameSniff.php:47]: Please remove the ignore annotation. This is testable. It just isn't tested (yet).

**Review by jrfnl — COMMENTED** [review]
LGTM. Happy to accept this PR once the ignore annotation has been removed.

I have two "pro-tips" related to this PR to keep in mind for the future:
1. **When test case files can't comply with the naming conventions**
    For this PR only one extra file which couldn't comply with the test case file naming convention needed to be added. In which case, adding the file the way it is done in this PR is perfectly fine.
    However, if you run across a sniff in the future, which needs multiple extra test case files which can't follow the `SniffNameUnitTest.#.inc` test case file name convention, things can get messy.
    So, my **pro-tip** for that case would be to create a sub-directory in the relevant test directory based on the sniff name and put all the additional test case files in that subdirectory.
    In the test file `getTestFiles()` method, you can then use a call to `glob()` to get a file list to return to the test runner.
    Unfortunately, I couldn't find an example of such a setup in this GH org, but you can see [an example of how this pattern can be applied in the WPCS standard for their `FileName` sniff](https://github.com/WordPress/WordPress-Coding-Standards/tree/develop/WordPress/Tests/Files).
    Hope that helps.
2. **Testing `STDIN`**
    It is perfectly possible to test code handling `STDIN`, so code blocks related to that should _never_ be ignored for code coverage.
    Yes, it does not fit within the "normal" testing framework for testing sniffs, but that doesn't mean, it can't be done.
    What would be needed for this, is to introduce a new test method to the test file. This may not be common, but there is nothing technically blocking this and the `DummyFile` class can be used to pass the input code to such tests.
    The new test method would then look something like this:
    ```php
    use PHP_CodeSniffer\Files\DummyFile;
    use PHP_CodeSniffer\Ruleset;
    use PHP_CodeSniffer\Tests\ConfigDouble;

    /*...*/

    /**
     * Test stdin.
     *
     * @return void
     */
    public function testStdIn()
    {
        $config            = new ConfigDouble();
        $config->standards = ['Generic'];
        $config->sniffs    = ['Generic.Files.LowercasedFilename'];

        $ruleset = new Ruleset($config);

        $content = '<?php '.PHP_EOL;
        $file    = new DummyFile($content, $ruleset, $config);
        $file->process();

        $this->assertSame(0, $file->getErrorCount());
        $this->assertSame(0, $file->getWarningCount());
        $this->assertCount(0, $file->getErrors());

    }//end testStdIn()

    ```
    And such a test could even be set up with a data provider to test multiple different code snippets and expectations.
    For some examples of test methods using a similar pattern, have a look at the [`ErrorSuppressionTest`](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/tests/Core/ErrorSuppressionTest.php).

**Comment by rodrigoprimo** [comment]
I forgot to mention in the previous comment that when implementing the test, I removed `PHP_EOL` from the contents of the dummy file as it seems it is not necessary. I am highlighting this here in case I missed something, and there is a reason to keep `PHP_EOL`.

**Comment by jrfnl** [comment]
> I forgot to mention in the previous comment that when implementing the test, I removed `PHP_EOL` from the contents of the dummy file as it seems it is not necessary. I am highlighting this here in case I missed something, and there is a reason to keep `PHP_EOL`.

All good, I just posted a rough example. Presumed you would customize it to what was needed.

**Review by jrfnl — APPROVED** [review]
Thanks @rodrigoprimo!

**Pull request merged**